src/Microsoft.Health.Dicom.Core/Features/Retrieve/DicomFileExtensions.cs

```diff
@@ -64,4 +69,27 @@ public static void ValidateFrames(DicomPixelData pixelData, IEnumerable<int> fra
             throw new FrameNotFoundException();
         }
     }
+
+    /// <summary>
+    /// Given a dicom file, the method will return the length of the dataset.
+    /// </summary>
+    /// <param name="dcmFile"> Dicom file</param>
+    /// <param name="recyclableMemoryStreamManager">RecyclableMemoryStreamManager to get Memory stream</param>
+    /// <returns>Dataset size</returns>
+    public static async Task<long> GetByteLengthAsync(DicomFile dcmFile, RecyclableMemoryStreamManager recyclableMemoryStreamManager)
```

Comment:
nit: extension method should have `this` keyword:
```suggestion
    public static async Task<long> GetByteLengthAsync(this DicomFile dcmFile, RecyclableMemoryStreamManager recyclableMemoryStreamManager)
```